tests/sql/distinct.slt

```diff
@@ -0,0 +1,19 @@
+statement ok
+CREATE TABLE test (x INT, y INT);
+
+statement ok
+INSERT INTO test VALUES (1, 1), (2, 2), (1, 1), (3, 3);
+
+query II
+SELECT DISTINCT * FROM test ORDER BY x;
+----
+1 1
+2 2
+3 3
+
+query I
+SELECT DISTINCT x FROM test ORDER BY y DESC;
```

Comment:
This is problematic. (Should be banned) e.g.,

```sql
> CREATE TABLE test (x INT, y INT);
created
in 0.001s
> insert into test values (1,3),(2,2),(3,1),(1,4);
4 rows inserted
in 0.001s
> select distinct x from test order by y;
+---+
| x |
+---+
| 3 |
| 2 |
| 1 |
| 1 |
+---+
in 0.003s
```

In postgres:
```sql
postgres=# SELECT DISTINCT x FROM test ORDER BY y DESC;
ERROR:  for SELECT DISTINCT, ORDER BY expressions must appear in select list
LINE 1: SELECT DISTINCT x FROM test ORDER BY y DESC;
  
```

Reply:
We don't need to ban it. Because we will add `Y` to the select_lists in binder. The previous implementation did have a bug. Now, the initial select list will be grouped(column x), and a `first` aggregation function will be added to the Y(added by OrderBy).

Reply:
Is there any DB support this behavior? IMO it's an unusal behavior, and is like select a column not in `GROUP BY`, which should be explicitly and carefully used when needed. So we should ban it :)

Reply:
I found PG has a special syntax [DISTINCT ON](https://www.postgresql.org/docs/current/sql-select.html#SQL-DISTINCT) for this behavior. Anyway, I would say this fancy feature is completely optional and we don't need to support it in the first place.